### PR TITLE
Improve the toFile documentation about file vs. string output

### LINKED ISSUE #502: Output to HTML string omit header / footer
I'm using AsciidoctorJ 1.5.4.1 and recently switched from calling `convertFile` with

    OptionsBuilder.options()
      .backend(backend)
      .docType(DOCTYPE)
      .eruby(ERUBY)
      .safe(SafeMode.UNSAFE)
      .baseDir(basedir)
      .toFile(outFile);

to calling it with

    ...
      .toFile(false);

I.e. I simply switch to outputting to a `String` instead an output file. While investigating the HTML output String I noticed it lack the HTML header / footer, meaning there is no `<!DOCTYPE html>` and `<head></head>` tag in there, and no `<div id="footer-text">` etc. in the end. I don"t believe this is expected as I did not see that behavior documented anywhere.

### PR DESCRIPTION
**Pull request by sschuberth**
Closes #502.

### CODE REVIEW

File: asciidoctorj-core/src/main/java/org/asciidoctor/Options.java [hunk 1]
```diff
@@ -67,10 +67,26 @@ public void setTemplateEngine(String templateEngine) {
...
+     *
+     */
     public void setToFile(String toFile) {
         this.options.put(TO_FILE, toFile);
     }
 
+    /**
+     * Toogle writing output to a file.
+     *
+     * @param toFile If <code>true</true>, overwrite the input file with the output,
```
Comment (robertpanzer) [inline asciidoctorj-core/src/main/java/org/asciidoctor/Options.java]: Would be great to have in here that this just changes the default of the headerFooter option and that it still can be changed.
Like this it reads that it's a fixed behavior creating an embeddable document with toFile=false and a full doc with toFile=true.

Comment (mojavelinux) [inline asciidoctorj-core/src/main/java/org/asciidoctor/Options.java]: It's also not true that it "overwrite(s) the input file with the output". It writes to a file adjacent to the input file which has the same root name and a file extension as specified by the converter.

**Comment by sschuberth** [comment]
@mojavelinux Sorry for the delay, I've updated the PR to address your comments.

**Comment by mojavelinux** [comment]
Thanks! Looks good to me!

**Comment by robertpanzer** [comment]
Thank you!

Cherry-picked into asciidoctorj-1.6.0 as 07fed407fce84dc9c0f7cf203eb8266255ebad7b

**Pull request merged**